src/components/dropdown/dropdown.tsx

```diff
@@ -0,0 +1,192 @@
+  disabled?: boolean;
+  multiple?: boolean;
+  options?: OptionProps[];
+  optionsGroup?: OptionGroupProps[];
+  selectedOptions?: OptionProps[];
+  size?: SelectSizes;
+  onSelectedOption: Function;
+}
+
+export interface OptionsConfig extends DropdownProps {
+  indexGroup?: number;
+}
+
+const isDisabled = (option: OptionProps, disabled?: boolean): boolean => {
+  if (disabled || option.disabled) return true;
+  return false;
+};
+const isSelectingNotMultiple = (
+  option: OptionProps,
+  multiple?: boolean
+): boolean => {
+  if (!option.selected && !multiple) return true;
+  return false;
+};
+
+const Options = ({
+  disabled,
+  options = [],
+  multiple,
+  onSelectedOption,
+  indexGroup,
+}: OptionsConfig) => {
+  const [optionsCopy, setOptionsCopy] = useState<OptionProps[]>(options);
+
+  const select = (index: number): void => {
+    if (isDisabled(optionsCopy[index], disabled)) return;
+    if (isSelectingNotMultiple(optionsCopy[index], multiple)) {
+      optionsCopy.forEach((option) => {
+        option.selected = false;
+      });
+    }
+    updateSelecteds(index);
+  };
+  const updateSelecteds = (index: number): void => {
+    optionsCopy[index].selected = !optionsCopy[index].selected;
+    setOptionsCopy([...optionsCopy]);
+    onSelectedOption([...optionsCopy]);
+  };
+
+  return (
+    <ul
+      data-testid={
+        indexGroup
+          ? 'ion-dropdown-options-' + indexGroup
+          : 'ion-dropdown-options'
+      }
+    >
+      {options &&
+        options.map((option, index) => {
+          return (
+            <OptionStyle
+              key={option.value}
+              data-testid={
+                indexGroup
+                  ? 'ion-option-' + index + '-' + indexGroup
+                  : 'ion-option-' + index
+              }
+              disabled={disabled || option.disabled}
+              selected={option.selected}
+              onClick={() => select(index)}
+            >
+              <div className="label">{option.label}</div>
+              {option.selected === true && (
+                <>
+                  <div className="icon check">
+                    <IonIcon type={'check'} size={20} />
+                  </div>
+                  <div className="icon close">
+                    <IonIcon type={'close'} size={20} />
+                  </div>
+                </>
+              )}
+            </OptionStyle>
+          );
+        })}
+    </ul>
+  );
+};
+
+const OptionsGroup = ({
+  disabled,
+  optionsGroup = [],
+  onSelectedOption,
+  multiple,
+}: DropdownProps) => {
+  const [groupCopy, setGroupCopy] = useState<OptionGroupProps[]>(optionsGroup);
+  const handleSelect = (event: OptionProps[]) => {
+    const isSelected = (optionToCompare: OptionProps): boolean => {
+      let result = false;
+      event.forEach((option) => {
+        if (option.selected && optionToCompare.value === option.value) {
+          result = true;
+        }
+      });
+      return result;
+    };
```

Comment:
I believe that is better this fuction out this scope

Reply:
In this case I'm not tryna filter selected options, or something like this, I'm just updating the same array. The  unselected options continue existing on it.